javascript/reactjs-todo-davinci/client/components/davinci-client/hooks/useDavinci.js

```diff
@@ -0,0 +1,139 @@
+/*
+ * ping-sample-web-react-davinci
+ *
+ * useDavinci.js
+ *
+ * Copyright (c) 2025 Ping Identity Corporation. All rights reserved.
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+import { useEffect, useMemo, useState } from 'react';
+import createClient from './create-client.utils.js';
+import useOAuth from './useOauth.js';
+
+/**
+ * @function useDavinci - Custom React hook that manages the DaVinci flow state
+ * @returns {Object} - An array with state at index 0 and setter methods at index 1
+ */
+export default function useDavinci() {
+  // Create local state for authorization
+  const [{ user }, { setCode }] = useOAuth();
```

Comment:
Non-blocking: I don't mind this as it's mimicking what we did in the original React sample app, but to be honest, I did like how you how this separated out and used `useOAuth` at the `form.js` level. I'm guessing you had a reason to move the use of `useOAuth` here?

Reply:
I think initially I was calling the oauth hook's `setCode` method in `useDavinci` then its `user` state in the form component. But that created two separate instances of state and so the user didn't get updated.

I suppose if you wanted useOauth called in form then useDavinci would have a dependency on it's setCode method. Something like:
```
  const [user, setCode] = useOAuth();
  const [{ formName, formAction, node, collectors }, { setNext, startNewFlow, updater }] =
    useDavinci(setCode);
```